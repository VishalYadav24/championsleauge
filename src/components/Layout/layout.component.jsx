import { Stack } from "@mui/system";
import React from "react";
import { Outlet } from "react-router-dom";
import Navbar from "../Navbar/navbar.component";

const Layout = () => {
  return (
    <Stack spacing={10}>
      <Navbar></Navbar>
      <Outlet />
    </Stack>
  );
};

export default Layout;
