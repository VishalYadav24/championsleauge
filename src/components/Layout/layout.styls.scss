@import '../../styles/variables.styles.scss';

.MuiStack-root.layout{
    background-color: $primary;
}
.content_container{
    height: calc(100vh - (13.063rem ));
    overflow: auto;
}

/* width */
::-webkit-scrollbar {
    width: 10px;
  }
  
  /* Track */
  ::-webkit-scrollbar-track {
    background: #f1f1f1;
  }
  
  /* Handle */
  ::-webkit-scrollbar-thumb {
    background: #888;
  }
  
  /* Handle on hover */
  ::-webkit-scrollbar-thumb:hover {
    background: #555;
  }