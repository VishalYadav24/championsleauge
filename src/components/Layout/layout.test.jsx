import { render, screen } from "@testing-library/react"
import { BrowserRouter } from "react-router-dom";
import Layout from "./layout.component";
import renderer from "react-test-renderer";



describe("Render Basic Layout of application", ()=>{
    test("<Layout/>",()=>{
        const layout = render(<Layout/>,{wrapper:BrowserRouter});
        expect(screen.getAllByTestId('layout')[0]).toBeInTheDocument();
    });

    test("<Layout/>Snapshot", () => {
        const layout = renderer
          .create(
            <BrowserRouter>
              <Layout  />
            </BrowserRouter>
          )
          .toJSON();
        expect(layout).toMatchSnapshot();
      });
})