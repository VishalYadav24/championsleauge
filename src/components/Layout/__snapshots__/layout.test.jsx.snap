// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Basic Layout of application <Layout/>Snapshot 1`] = `
<div
  className="MuiStack-root layout css-1uhwev9-MuiStack-root"
  data-testid="layout"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <ul
        className="MuiList-root MuiList-padding list_container css-h4y409-MuiList-root"
      >
        <li
          className="MuiListItem-root MuiListItem-gutters MuiListItem-padding list_item_1 css-1p823my-MuiListItem-root"
          disabled={false}
        >
          <button
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium home css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <img
              alt="logo"
              className="logo"
              src="swords.png"
            />
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <h3>
            Game Store
          </h3>
        </li>
        <li
          className="MuiListItem-root MuiListItem-gutters MuiListItem-padding list_item_2 css-1p823my-MuiListItem-root"
          disabled={false}
        >
          <button
            aria-label="Watchlist"
            aria-labelledby={null}
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium watchlist css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            data-mui-internal-clone-element={true}
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            placeholder="bottom"
            tabIndex={0}
            type="button"
          >
            <img
              alt="watch list"
              className="logo"
              src="bookmark.png"
            />
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
      </ul>
    </div>
  </header>
</div>
`;
